sdmetrics/reports/single_table/diagnostic_report.py

```diff
@@ -0,0 +1,265 @@
+            if np.isnan(score):
+                continue
+            if score >= 0.9:
+                self._results['SUCCESS'].append(RESULT_DETAILS[metric]['SUCCESS'])
+            elif score >= 0.5:
+                self._results['WARNING'].append(RESULT_DETAILS[metric]['WARNING'])
+            else:
+                self._results['DANGER'].append(RESULT_DETAILS[metric]['DANGER'])
+
+        out.write('DiagnosticResults:\n')
+        self._print_results_for_level(out, 'SUCCESS')
+        self._print_results_for_level(out, 'WARNING')
+        self._print_results_for_level(out, 'DANGER')
+
+    def generate(self, real_data, synthetic_data, metadata):
+        """Generate report.
+
+        Args:
+            real_data (pandas.DataFrame):
+                The real data.
+            synthetic_data (pandas.DataFrame):
+                The synthetic data.
+            metadata (dict):
+                The metadata, which contains each column's data type as well as relationships.
+        """
+        metrics = list(itertools.chain.from_iterable(self.METRICS.values()))
+
+        for metric in tqdm.tqdm(metrics, desc='Creating report'):
+            metric_name = metric.__name__
+            try:
+                self._metric_results[metric_name] = metric.compute_breakdown(
+                    real_data, synthetic_data, metadata)
+
+                if 'score' in self._metric_results[metric_name]:
+                    self._metric_averages[metric_name] = self._metric_results[metric_name]['score']
+                else:
+                    avg_score, _ = aggregate_metric_results(self._metric_results[metric_name])
+                    self._metric_averages[metric_name] = avg_score
+
+            except IncomputableMetricError:
+                # Metric is not compatible with this dataset.
+                self._metric_results[metric_name] = {}
+
+        self._print_results()
+
+    def get_results(self):
+        """Return the diagnostic results.
+
+        Returns:
+            dict
+                The diagnostic results.
+        """
+        return copy.deepcopy(self._results)
+
+    def get_properties(self):
+        """Return the property score breakdown.
+
+        Returns:
+            pandas.DataFrame
+                The property score breakdown.
+        """
+        properties = {}
+        for prop, metrics in self.METRICS.items():
+            prop_scores = [self._metric_averages[metric.__name__] for metric in metrics]
+            properties[prop] = np.nanmean(prop_scores)
+
+        return properties
+
+    def get_visualization(self, property_name):
+        """Return a visualization for each score for the given property name.
+
+        Args:
+            property_name (str):
+                The name of the property to return score details for.
+
+        Returns:
+            plotly.graph_objects._figure.Figure
+                The visualization for the requested property.
+        """
```

Comment:
can we add pass or return back in here?

Reply:
I'm going to leave it out for now since the new PR will be merged shortly anyway.